### PR TITLE
Add bitcoin-cli man page

### PR DESCRIPTION
**Pull request by Ciemon**
Initial man page for bitcoin-cli

### CODE REVIEW

File: contrib/debian/manpages/bitcoin-cli.1 [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+\fB\-rpcssl\fR=\fI1\fR
+Use OpenSSL (https) for JSON\-RPC connections
+.TP
+\-?
+This help message
+
+.SH "SEE ALSO"
+bitcoind.d(1), bitcoin.conf(5)
+.SH AUTHOR
+This manual page was written by Ciemon Dunville. Permission is granted to copy, distribute and/or modify this document under the terms of the GNU General Public License, Version 3 or any later version published by the Free Software Foundation.
```
Comment (laanwj) [inline contrib/debian/manpages/bitcoin-cli.1]: Thanks for adding a manpage for bitcoin-cli! However, can you please license it under the same license as the rest of the repository? (which is the MIT license, see COPYING)

**Comment by luke-jr** [comment]
Will this man page get installed as part of a Debian package at least? I would expect some other file needs updating for that... but I could be wrong.

File: contrib/debian/manpages/bitcoin-cli.1 (hunk 1 above)
Comment (Ciemon) [inline contrib/debian/manpages/bitcoin-cli.1]: I could, but I've used the same license as the other Debian man pages, and
as I'm sure you know master/contrib/debian/copyright is fairly big already;
arguably it doesn't need me dropping more lines into it?

File: contrib/debian/manpages/bitcoin-cli.1 [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+.SH NAME
+bitcoin-cli \- the remote procedure call client for bitcoin core. 
+.SH SYNOPSIS
+bitcoin-cli [options] <command> [params] \- Send command to Bitcoin Core. 
+.TP
+bitcoin-cli [options] help \- List Bitcoin Core commands.
+.TP
+bitcoin [options] help <command> \- Get help for a command.
+.SH DESCRIPTION
+This manual page documents the bitcoin-cli program. bitcoin-cli is the RPC client used to send commands to Bitcoin Core.
```
Comment (sipa) [inline contrib/debian/manpages/bitcoin-cli.1]: I would say: "Is *an* RPC client" or even "Is a minimal RPC client". Using "the" seems to imply it's only one.

File: contrib/debian/manpages/bitcoin-cli.1 [hunk 3]
```diff
@@ -0,0 +1,52 @@
+.TH BITCOIN-CLI "1" "February 2015" "bitcoin-cli 0.10" 
+.SH NAME
+bitcoin-cli \- the remote procedure call client for bitcoin core. 
+.SH SYNOPSIS
+bitcoin-cli [options] <command> [params] \- Send command to Bitcoin Core. 
+.TP
+bitcoin-cli [options] help \- List Bitcoin Core commands.
```
Comment (sipa) [inline contrib/debian/manpages/bitcoin-cli.1]: Maybe: Ask Bitcoin Core for a list of supported commands (to indicate that the result of this command depends on the server, and not the client).

File: contrib/debian/manpages/bitcoin-cli.1 [hunk 4]
```diff
@@ -0,0 +1,52 @@
+.TH BITCOIN-CLI "1" "February 2015" "bitcoin-cli 0.10" 
+.SH NAME
+bitcoin-cli \- the remote procedure call client for bitcoin core.
```
Comment (sipa) [inline contrib/debian/manpages/bitcoin-cli.1]: Can you use consistent capitalization for Bitcoin Core?

File: contrib/debian/manpages/bitcoin-cli.1 [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+.SH OPTIONS
+.TP
+\fB\-conf=\fR<file>
+Specify configuration file (default: bitcoin.conf)
+.TP
+\fB\-datadir=\fR<dir>
+Specify data directory
+.TP
+\fB\-testnet\fR
+Use the test network
```
Comment (sipa) [inline contrib/debian/manpages/bitcoin-cli.1]: Maybe: Connect to a Bitcoin Core instance running in testnet mode.

File: contrib/debian/manpages/bitcoin-cli.1 [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+.TP
+\fB\-regtest\fR
+Enter regression test mode, which uses a special chain in which blocks can be solved instantly. This is intended for regression testing tools and app development.
```
Comment (sipa) [inline contrib/debian/manpages/bitcoin-cli.1]: Maybe: Connect to a Bitcoin Core instance running in regtest mode (see documentation for -regtest on bitcoind).

**Comment by Ciemon** [comment]
As comments have now died down and all changes have been made this man page is now complete.

**Comment by fanquake** [comment]
@Ciemon You'll need to squash your commits.

**Comment by Ciemon** [comment]
Unfortunately I seem to have reached the limit of my capability with this. I'm unable to squash the commits; I can't work out how to do it.

**Comment by paveljanik** [comment]
Squashing is described e.g. here:

http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Do not hesitate to ask on IRC.

**Comment by ajweiss** [comment]
Is rpcwait being intentionally deprecated/hidden?

**Comment by Ciemon** [comment]
No Adam, thanks for noticing that I'd missed it, you can't beat a fresh set
of eyes.

**Comment by ajweiss** [comment]
Have a look at #5821.

Some sort of note that advises that testing server connectivity with
"-rpcwait" may lead to frustration (as it just silently waits for the
server to appear, potentially forever) might be a good idea.

--adam

**Comment by laanwj** [comment]
I'll squash/rebase  and merge this for you this time. I'd suggest to try practicing a bit with simple changes in git, without involving github.

**Comment by laanwj** [comment]
Merged via 96c19a3

**Pull request closed without merging**